frontend/src/editor/indent.ts

```diff
@@ -0,0 +1,189 @@
+/* eslint-disable no-param-reassign */
+// Sources:
+// https://github.com/ueberdosis/tiptap/issues/1036#issuecomment-1000983233
+// https://github.com/ueberdosis/tiptap/issues/1036#issuecomment-981094752
+// https://github.com/django-tiptap/django_tiptap/blob/main/django_tiptap/templates/forms/tiptap_textarea.html#L453-L602
+
+import {
+  CommandProps,
+  Extension,
+  Extensions,
+  isList,
+  KeyboardShortcutCommand,
+} from "@tiptap/core"
+import { TextSelection, Transaction } from "prosemirror-state"
+
+declare module "@tiptap/core" {
+  interface Commands<ReturnType> {
+    indent: {
+      indent: () => ReturnType
+      outdent: () => ReturnType
+    }
+  }
+}
+
+type IndentOptions = {
+  names: string[]
+  indentRange: number
+  minIndentLevel: number
+  maxIndentLevel: number
+  defaultIndentLevel: number
+  HTMLAttributes: Record<string, unknown>
+}
+
+type IndentType = "indent" | "outdent"
+
+const clamp = (val: number, min: number, max: number): number =>
+  Math.min(Math.max(val, min), max)
+
+const updateIndentLevel = (
+  tr: Transaction,
```

Comment:
I personally often have difficulties with abbreviations in code. At first I thought: "What does table row have to do with this?". What do you think about renaming this to "transaction". I think it is not too long.